File: src/main/java/de/poeschl/bukkit/logcleaner/managers/SettingManager.java
package de.poeschl.bukkit.logcleaner.managers;

import org.bukkit.configuration.file.FileConfiguration;

import java.util.logging.Logger;


public class SettingManager {

    private static final String KEEP_DAYS_KEY = "keepLogsFromLastDays";

    private FileConfiguration config;
    private Logger logger;

    public SettingManager(FileConfiguration config, Logger logger) {
        this.config = config;
        this.logger = logger;
    }

    public void updateConfig(FileConfiguration config) {
        this.config = config;
    }

    public int getKeepDays() {
        return config.getInt(KEEP_DAYS_KEY);
    }
}

File: src/main/kotlin/de/poeschl/bukkit/logcleaner/managers/SettingManager.kt
package de.poeschl.bukkit.logcleaner.managers

import de.poeschl.bukkit.logcleaner.managers.SettingsManager.Companion.KEEP_DAYS_KEY
import org.bukkit.configuration.file.FileConfiguration

import java.util.logging.Logger

interface SettingsManager {

    val keepDays: Int

    fun updateConfig(config: FileConfiguration)

    companion object {
        val KEEP_DAYS_KEY = "keepLogsFromLastDays"
    }
}

class SettingsManagerImpl(private var config: FileConfiguration, private val logger: Logger) : SettingsManager {

    override val keepDays: Int
        get() {
            return config.getInt(KEEP_DAYS_KEY)
        }

    override fun updateConfig(config: FileConfiguration) {
        this.config = config
    }
}
